File: src/main/java/de/ng/ahnenwin2gedcom/csv/Geschlecht.java
package de.ng.ahnenwin2gedcom.csv;

import java.util.Arrays;
import java.util.List;

public enum Geschlecht {
    MAENNLICH("m", "M"),
    WEIBLICH("w", "W"),
    UNBEKANNT();
    private final String[] csvValues;

    Geschlecht(String... csvValues) {
        this.csvValues = csvValues;
    }

    static Geschlecht findByCsvValue(String csvValue) {
        String trimmedCsvValue = csvValue == null ? "" : csvValue.trim();
        return Arrays.stream(values())
                .filter(geschlecht -> List.of(geschlecht.csvValues).contains(trimmedCsvValue))
                .findAny()
                .orElse(Geschlecht.UNBEKANNT);
    }
}

File: src/main/java/de/ng/ahnenwin2gedcom/hej/Geschlecht.kt
package de.ng.ahnenwin2gedcom.hej

enum class Geschlecht(private vararg val csvValues: String) {
    MAENNLICH("m", "M"),
    WEIBLICH("w", "W"),
    UNBEKANNT;

    internal companion object {
        @JvmStatic
        fun findByCsvValue(csvValue: String?): Geschlecht {
            return values().find { csvValue?.trim() in it.csvValues} ?: UNBEKANNT
        }
    }

    fun opposite(): Geschlecht {
        return when(this) {
            MAENNLICH -> WEIBLICH
            WEIBLICH -> MAENNLICH
            UNBEKANNT -> UNBEKANNT
        }
    }

}
